reference/configuration/framework.rst

```diff
@@ -169,39 +293,431 @@ see :doc:`/cookbook/request/load_balancer_reverse_proxy`.
+
+collect
+.......
+
+.. versionadded:: 2.3
+    The ``collect`` option was introduced in Symfony 2.3. Previously, when
+    ``profiler.enabled`` was ``false``, the profiler *was* actually enabled,
+    but the collectors were disabled. Now, the profiler and the collectors
+    can be controlled independently.
+
+**type**: ``boolean`` **default**: ``true``
+
+This option configures the way the profiler behaves when it is enabled. If set
+to ``true``, the profiler collects data for all requests. If you want to only
+collect information on-demand, you can set the ``collect`` flag to ``false``
+and activate the data collectors manually::
+
+    $profiler->enable();
+
+only_exceptions
+...............
+
+**type**: ``boolean`` **default**: ``false``
+
+When this is set to ``true``, the profiler will only be enabled when an
+exception is thrown during the handling of the request. It becomes the service
+container parameter called ``profiler_listener.only_exceptions``.
+
+only_master_requests
+....................
+
+**type**: ``boolean`` **default**: ``false``
+
+When this is set to ``true``, the profiler will only be enabled on the master
+requests (and not on the subrequests). It becomes the service container
+parameter called ``profiler_listener.only_master_requests``.
+
+dsn
+...
+
+**type**: ``string`` **default**: ``'file:%kernel.cache_dir%/profiler'``
+
+The DSN where to save the profiling information. The supported drivers (part
+before the colon) are:
+
+* file
+* sqlite
+* mysql
+* mongodb
+* memcache
+* memcached
+* redis
+
+It becomes the service container parameter called ``profiler.storage.dsn``.
+
+username
+........
+
+**type**: ``string`` **default**: ``''``
+
+When needed, the username for the profiling storage. It becomes the service
+container parameter called ``profiler.storage.username``.
+
+password
+........
+
+**type**: ``string`` **default**: ``''``
+
+When needed, the password for hte profiling storage. It becomes the service
+container parameter called ``profiler.storage.password``.
+
+lifetime
+........
+
+**type**: ``integer`` **default**: ``86400``
+
+The lifetime of the profiling storage in seconds. The data will be deleted when
+the lifetime is expired. It becomes the service container parameter called
+``profiler.storage.lifetime``.
+
+
+matcher
+.......
+
+.. seealso::
+
+    See :doc:`/cookbook/profiler/matchers` for more information about using
+    matchers to enable/disable the profiler.
+
+ip
+""
+
+**type**: ``string``
+
+If set, the profiler will only be enabled when the current IP address matches.
+
+.. _reference-profiler-matcher-path:
+
+path
+""""
+
+**type**: ``string``
+
+If set, the profiler will only be enabled when the current path matches.
+
+service
+"""""""
+
+**type**: ``string``
+
+This setting contains the service id of a custom matcher.
+
+router
+~~~~~~
+
+resource
+........
+
+**type**: ``string`` **required**
+
+Specifies the path to the routes used by the default router.
```

Comment:
How about something like:

>
The path the main routing resource (e.g. a YAML file) that contains the routes and imports the router should load.